### DISCUSSION TITLE
Livewire security issue

### DISCUSSION THREAD

**Discussion by carlosoberto88** [livewire/livewire#5119] [1. Help]
Hello, 

I have a security issue in Livewire, users are able to make post request via postman to the url Livewire/message/XXXXX

They are sending the cookies and the CSRF token found in the website, is there any way I can stop them from doing this? maybe add a middleware? or other security measure?
![evidence](https://user-images.githubusercontent.com/19641157/180231384-a6e1179c-6df1-422d-b996-b8ab769dcd3e.png)

**Comment by joshhanley** [comment]
@carlosoberto88 Livewire actually has some extra security measures built in, to stop people tampering with the Livewire requests. See the security section in the docs https://laravel-livewire.com/docs/2.x/security#security-measures

But you won't be able to stop people hitting the end point, no different if you had a POST route in your app and someone kept sending requests to that route.

That's why it's also important to make sure you have proper authorisation in your controllers and Livewire components, to ensure everything is properly secure.

If you have any further concerns or questions, please let me know and I'd be happy to answer them 🙂

Hope this helps!

**Reply by carlosoberto88** [reply]
Thanks for your answer! just to make sure, will they be able to access some information about other users in the endpoint?

**Reply by joshhanley** [reply]
@carlosoberto88 the endpoint itself actually has no information at all, other than to load the correct component specified.

But even if a user managed to get the correct parameters to send to Livewire's endpoint to request a specific component, if the request hasn't originated from the component, it won't have a valid checksum, as such will display this error

![image](https://user-images.githubusercontent.com/882837/181123351-72095afb-4b5b-4719-9aa5-95410c4f1db3.png)

That being said, any methods or actions that are available in your component, lets say they did manage to get a valid request and checksum (which would only be possible if that are already logged in and acting as their own user anyway), should still have proper authorisation in your components to ensure that user has permission to do what they are trying to do. No different to a Laravel controller endpoint.

To do this, you would use the normal authorisation processes, see here https://laravel-livewire.com/docs/2.x/authorization

Hope this helps!

**Reply by Gishi2** [reply]
Hello @joshhanley ,

I hope it is not too late for me to still ask questions related to livewire security issues. Im still a beginner in laravel and software development and same as this post, im doing a task that is related to livewire security issues as well.

Similar to this post, making a post request through an external such as postman, you quoted that,

> That's why it's also important to make sure you have proper authorisation in your controllers and Livewire components, to ensure everything is properly secure.

Just because it is not clear to me yet, by adding proper authorisations in to our controller and livewire components, we can block requests that comes from external, is it like that?

My concern on a specific level is when someone authorized or unauthorized makes a post request to upload file using livewire endpoint, _livewire/upload-file_,  because i added a restriction on file format which works on the frontend, but when i did a little bit of testing of post request via postman, using file format that should be restricted, it returns 200 OK. So im trying to figure out on how to restrict external post request.
